### PR TITLE
bridge driver: various code quality improvements

### PR DESCRIPTION
**Pull request by rhansen**
**- What I did**

This PR contains a handful of commits (please do not squash) that modify the bridge driver to improve code readability and maintainability.

These commits are preparation for [my work-in-progress fix](https://github.com/rhansen/moby/commit/host_ipv6) for #46469.

The first commit was suggested in https://github.com/moby/moby/pull/46446#discussion_r1321377818

No behavior changes are expected, but I'm not 100% certain that the last commit (the one that renames `HostIP` to `HostIPv4`) preserves API and disk file format compatibility.  I'm OK with omitting that commit, but I think it improves readability, especially when combined with the fix for #46469.

**- Description for the changelog**

(not necessary)

### CODE REVIEW

**Review by neersighted — COMMENTED** [review]
This looks overall quite positive to me, though I'd like to confirm that on-disk compatibility is kept for myself before a final LGTM.

File: libnetwork/drivers/bridge/setup_ip_tables_linux.go [hunk 1]
```diff
@@ -200,140 +196,138 @@ func (n *bridgeNetwork) setupIPTables(ipVersion iptables.IPVersion, maskedAddr *
...
+}
+
+// Append appends the rule to the end of the chain. If the rule already exists anywhere in the
+// chain, this is a no-op.
+func (r iptRule) Append() error {
+	if r.Exists() {
+		return nil
+	}
+	return r.exec("-A")
 }
```
Comment (corhere) [inline libnetwork/drivers/bridge/setup_ip_tables_linux.go:225]: Shockingly, we already have a newtype and constants for the iptables operations defined. May as well use them.

```suggestion
func (r iptRule) cmdArgs(op iptables.Action) []string {
	return append([]string{"-t", string(r.table), string(op), r.chain}, r.args...)
}

func (r iptRule) exec(op iptables.Action) error {
	return iptables.GetIptable(r.ipv).RawCombinedOutput(r.cmdArgs(op)...)
}

// Append appends the rule to the end of the chain. If the rule already exists anywhere in the
// chain, this is a no-op.
func (r iptRule) Append() error {
	if r.Exists() {
		return nil
	}
	return r.exec(iptables.Append)
}
```

File: libnetwork/drivers/bridge/bridge_store.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func (ncfg *networkConfiguration) MarshalJSON() ([]byte, error) {
 	nMap["Internal"] = ncfg.Internal
 	nMap["DefaultBridge"] = ncfg.DefaultBridge
 	nMap["DefaultBindingIP"] = ncfg.DefaultBindingIP.String()
-	nMap["HostIP"] = ncfg.HostIP.String()
+	nMap["HostIPv4"] = ncfg.HostIPv4.String()
```
Comment (corhere) [inline libnetwork/drivers/bridge/bridge_store.go:149]: That's an incompatible change to the on-disk format. Making it a compatible change is thankfully trivial:
```suggestion
	nMap["HostIP"] = ncfg.HostIPv4.String()
```

File: libnetwork/drivers/bridge/bridge_store.go [hunk 3]
```diff
@@ -189,8 +189,8 @@ func (ncfg *networkConfiguration) UnmarshalJSON(b []byte) error {
 		ncfg.ContainerIfacePrefix = v.(string)
 	}
 
-	if v, ok := nMap["HostIP"]; ok {
-		ncfg.HostIP = net.ParseIP(v.(string))
+	if v, ok := nMap["HostIPv4"]; ok {
```
Comment (corhere) [inline libnetwork/drivers/bridge/bridge_store.go:192]: Ditto.
```suggestion
	if v, ok := nMap["HostIP"]; ok {
```

**Review by corhere — CHANGES_REQUESTED** [review]
The last commit retains API compatibility as far as I can tell, but it does break on-disk compatibility as you feared. Luckily it's easy enough to fix.

Everything else looks good to me. The new `iptRule` API is such a massive improvement!

File: libnetwork/drivers/bridge/setup_ip_tables_linux.go (hunk 1 above)
Comment (rhansen) [reply libnetwork/drivers/bridge/setup_ip_tables_linux.go]: Done.

File: libnetwork/drivers/bridge/bridge_store.go (hunk 2 above)
Comment (rhansen) [reply libnetwork/drivers/bridge/bridge_store.go]: Done.

File: libnetwork/drivers/bridge/bridge_store.go (hunk 3 above)
Comment (rhansen) [reply libnetwork/drivers/bridge/bridge_store.go]: Done.

**Comment by rhansen** [comment]
> The last commit retains API compatibility as far as I can tell, but it does break on-disk compatibility as you feared. Luckily it's easy enough to fix.

Thanks for checking @corhere, and thanks for the review!

In addition to making the suggested changes, I rebased the branch onto the latest `master`.

**Review by neersighted — APPROVED** [review]

**Review by corhere — APPROVED** [review]
LGTM!

**Comment by corhere** [comment]
The only test failure is our old flaky friend TestNetworkDBCRUDTableEntries.
```
=== RUN   TestNetworkDBCRUDTableEntries
    networkdb_test.go:309: Entry existence verification test failed for node2(64b4fc3dcb43)
--- FAIL: TestNetworkDBCRUDTableEntries (8.22s)
```

**Pull request merged**